src/variable.cpp

```diff
@@ -15,23 +15,28 @@ limitations under the License.
 */
 #include "libcellml/variable.h"
 
+#include <algorithm>
+#include <cassert>
+#include <map>
 #include <sstream>
 #include <vector>
-#include <algorithm>
 
 #include "libcellml/units.h"
 
 
 namespace libcellml {
 
 /**
- * @brief Convert a INTERFACE_TYPES into its string form.
- * Private function to convert a Variable INTERFACE_TYPES into its string form.
- *
- * @param interfaceType The interface type to convert.
- * @return A @c std::string form of the given interface type.
+ * @brief Map to convert an interface type into its string form.
+ * An internal map used to convert a Variable InterfaceTypes enum class member into its string form.
  */
-EXPORT_FOR_TESTING std::string interfaceTypeToString(Variable::INTERFACE_TYPES interfaceType);
+std::map<Variable::InterfaceTypes, std::string> interfaceTypeToString =
+{
+    {Variable::InterfaceTypes::NONE, "none"},
+    {Variable::InterfaceTypes::PRIVATE, "private"},
+    {Variable::InterfaceTypes::PUBLIC, "public"},
+    {Variable::InterfaceTypes::PUBLIC_AND_PRIVATE, "public_and_private"}
```

Comment:
I think this might go against the [Google C++ style guide on braced initialiser lists](https://google.github.io/styleguide/cppguide.html#Braced_Initializer_List_Format)? (see the last example) But I can add an exception in our coding style and standardise across our braced lists if this is a strong preference.